controlplane/modules/route/internal/rib/map_trie.go

```
@@ -0,0 +1,70 @@
+package rib
+
+import (
+	"net/netip"
+)
+
+type MapTrieKey struct {
+	Prefix netip.Prefix
+}
+
+type MapTrie [129]map[MapTrieKey]*RoutesList
+
+func NewMapTrie(preAllocSize int) MapTrie {
+	mt := MapTrie{}
+	for idx := range mt {
+		mt[idx] = make(map[MapTrieKey]*RoutesList, preAllocSize)
+	}
+	return mt
+}
+
+func (m *MapTrie) Lookup(addr netip.Addr) (*RoutesList, bool) {
+	maxBits := 129
+	base := 128
+	if addr.Is4() {
+		base = 32
+		maxBits = 33
+	}
+	for n := range maxBits {
+		bits := base - n
+		p, err := addr.Prefix(bits)
```

If it is impossible, maybe just ignore error here?